x/oracle/keeper/oracle.go

```diff
@@ -31,7 +31,12 @@ func (k Keeper) RegisterOracle(ctx sdk.Context, msg *types.MsgRegisterOracle) er
 
 	// TODO: Add active queue of Oracle Registration
 
-	// TODO: emit RegisterOracle event
+	ctx.EventManager().EmitEvents(sdk.Events{
+		sdk.NewEvent(
+			types.EventTypeRegisterOracle,
+			sdk.NewAttribute(types.AttributeKeyOracle, types.AttributeValueOracle)),
+	})
```

Comment:
When you see the [guide](https://docs.cosmos.network/master/core/events.html), the attribute of event is to give more information about the categorized event.
So, we don't have to add a new attribute if we use a fixed value of attribute. I think this event is to alert all oracles no matter who register, we don't need a new attribute.

```suggestion
	ctx.EventManager().EmitEvents(sdk.Events{
		sdk.NewEvent(
			types.EventTypeRegisterOracle,
		),
	})
```

Reply:
As you said, it seems to work, but all events seem to respond to events in the form `{EventType}.{AttributeKey}={AttributeValue}`.
```json
  {
    "type":"coin_received",
    "attributes":[
      {
        "key":"cmVjZWl2ZXI=",
        "value":"cGFuYWNlYTE3eHBmdmFrbTJhbWc5NjJ5bHM2Zjg0ejNrZWxsOGM1bGFrbHhzNw==",
        "index":true
      },
      {
        "key":"YW1vdW50",
        "value":"MTAwMDAwMHVtZWQ=",
        "index":true
      }
    ]
  },
  {
    "type":"transfer",
    "attributes":[
      {
        "key":"cmVjaXBpZW50",
        "value":"cGFuYWNlYTE3eHBmdmFrbTJhbWc5NjJ5bHM2Zjg0ejNrZWxsOGM1bGFrbHhzNw==",
        "index":true
      },
      {
        "key":"c2VuZGVy",
        "value":"cGFuYWNlYTF6dHl5YTZ6cTduZzNtN2x2ejV4a3UzNmM0dDVqNzAyZ3A0YWc2YQ==",
        "index":true
      },
      {
        "key":"YW1vdW50",
        "value":"MTAwMDAwMHVtZWQ=",
        "index":true
      }
    ]
  }
```


If it is as suggested, I think the result will be answered as below.
```json
{
    "type": "register",
    "attributes": []
}
```

So, I think it would be good for us to respond as well.
```json
{
    "type": "vote",
    "attributes": [
        {
            "key": "voteType",
            "value": "registerOracle"
        }
    ]
}
```

What do you think?

Reply:
Oh I understood. I think the way @gyuguen suggested is better.